crates/pathfinder/src/rpc/v02/method/get_events.rs

```diff
@@ -0,0 +1,742 @@
+
+        let input = GetEventsInput {
+            filter: EventFilter {
+                from_block: None,
+                to_block: None,
+                address: None,
+                keys: keys_for_expected_events.clone(),
+                chunk_size: 2,
+                // TODO Some(0)
+                continuation_token: None,
+            },
+        };
+        let result = get_events(context.clone(), input).await.unwrap();
+        assert_eq!(
+            result,
+            GetEventsResult {
+                events: expected_events[..2].to_vec(),
+                continuation_token: Some(1.to_string()),
+            }
+        );
+
+        let input = GetEventsInput {
+            filter: EventFilter {
+                from_block: None,
+                to_block: None,
+                address: None,
+                keys: keys_for_expected_events.clone(),
+                chunk_size: 2,
+                continuation_token: Some(1.to_string()),
+            },
+        };
+        let result = get_events(context.clone(), input).await.unwrap();
+        assert_eq!(
+            result,
+            GetEventsResult {
+                events: expected_events[2..4].to_vec(),
+                continuation_token: Some(2.to_string()),
+            }
+        );
+
+        let input = GetEventsInput {
+            filter: EventFilter {
+                from_block: None,
+                to_block: None,
+                address: None,
+                keys: keys_for_expected_events.clone(),
+                chunk_size: 2,
+                continuation_token: Some(2.to_string()),
+            },
+        };
+        let result = get_events(context.clone(), input).await.unwrap();
+        assert_eq!(
+            result,
+            GetEventsResult {
+                events: expected_events[4..].to_vec(),
+                continuation_token: None,
+            }
+        );
+
+        // nonexistent page
+        let input = GetEventsInput {
+            filter: EventFilter {
+                from_block: None,
+                to_block: None,
+                address: None,
+                keys: keys_for_expected_events.clone(),
+                chunk_size: 2,
+                continuation_token: Some(3.to_string()),
+            },
+        };
+        let error = get_events(context, input).await.unwrap_err();
+        assert_eq!(error, GetEventsError::InvalidContinuationToken);
+    }
+
+    mod pending {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[tokio::test]
+        async fn backward_range() {
+            let context = RpcContext::for_tests_with_pending().await;
+
+            let input = GetEventsInput {
+                filter: EventFilter {
+                    from_block: Some(BlockId::Pending),
+                    to_block: Some(BlockId::Latest),
+                    address: None,
+                    keys: vec![],
+                    chunk_size: 100,
+                    // TODO Some(0)
```

Comment:
`TODO`

Reply:
Thanks, I forgot to test the variant in the TODO too.

Reply:
I added this variant to some of the tests.